src/lib/components/PreviewButton.js

```diff
@@ -63,7 +69,7 @@ export class PreviewButtonComponent extends Component {
       >
         {(formik) => (
           <>
-            {isLoading ? (
+            { ( formik.isSubmitting || isLoading ) ? (
```

Comment:
up for discussion: Interesting, so if a user clicks on "Save Draft", all other buttons (here we see the pattern on the PreviewButton) won't just be disabled, they will also be spinning? I would just make the clicked button spinning and the rest disabled. The feedback comes out quite strong otherwise and less specific.

Reply:
Hmm, currently if you click on the "Preview" button the "Save Draft" button will also be spinning and disabled (when we click the "Preview" button we also trigger the save) but the rest don't are disabled nor spin because the `formik.isSubmitting` is set to `False` after the "onSubmit()" (check [here](https://formik.org/docs/guides/form-submission)) but the redux state is still loading. This causes that when we click a button, the `formik.isSubmitting` changes to true for a very short period of time while it stays in the loading state for much more time (until we get the response). I agree with making the clicked buttons spin and have the others disabled but we would need to check for all the redux loading states to disable them since the `formik.isSubmitting` is set to false before redux state is updated. Would this check be OK? or any better ideas?

Reply:
> This causes that when we click a button, the formik.isSubmitting changes to true for a very short period of time

I think this should be solved by https://github.com/inveniosoftware/react-invenio-deposit/pull/257 now. There was indeed a bug where the `formik.isSubmitting` was very short, but now it should be as long as it takes to get the response. So other buttons should be disabled correctly now. Making `formik.isSubmitting && isLoading` the condition for spinning and `formik.isSubmitting` the condition for disabled here should be good, right?

Reply:
You are right, `formik.isSubmitting` bug is solved. Your proposal works flawless. Thanks a lot!